### PR TITLE
Backports misc. fixes for the new ofborg eval time checks

### PR DESCRIPTION
**Pull request by samueldr**
###### Motivation for this change

The new ofborg check (named grahamcofborg-eval-package-list-no-aliases) currently fails on release-18.09.

image

This PR aims to correct this.

###### Things done

- ✔️ Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - ✔️ NixOS
   - ⬜ macOS
   - ⬜ other Linux distributions
- ✔️ `time env -i nix-env --query --available --json --file . --arg config '{ allowAliases = false; }'`
- ✔️ Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @ryantm

in f91a79e there were changes to `pkgs/misc/vim-plugins/default.nix` which didn't seem to be needed. I have **not** backported those changes. Any explanations to help figure out whether they should have been needed?

### CODE REVIEW

**Comment by samueldr** [comment]
🎉 it eval'd!

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -21963,11 +21962,6 @@ with pkgs;
     callPackage ../applications/networking/cluster/terraform-providers {}
   );
 
-  terraform-provider-libvirt = callPackage ../applications/networking/cluster/terraform-provider-libvirt {};
-
-  terraform-provider-ibm = callPackage ../applications/networking/cluster/terraform-provider-ibm {};
-
-
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Where did those go?

Comment (samueldr) [reply pkgs/top-level/all-packages.nix]: Hmm, overlooked when asking @ryantm 

This is verbatim from the cherry-picked commit.

**Comment by samueldr** [comment]
This is merged without the equivalent change to the vim-plugins as that change is not required for fixing ofborg evals.

If the ofborg checks are updated to dig-in deeply in the attribute sets, then I guess it'll have to be ported, but in all cases, it's not a matter of backporting, there is a merge conflict.

**Pull request merged**